cocos/ui/UIScale9Sprite.cpp

```diff
@@ -105,15 +113,20 @@ namespace ui {
     bool Scale9Sprite::initWithSpriteFrameName(const std::string& spriteFrameName,
                                                const Rect& capInsets)
     {
-        CCASSERT((SpriteFrameCache::getInstance()) != NULL,
-                 "SpriteFrameCache::getInstance() must be non-NULL");
-
-        SpriteFrame *frame = SpriteFrameCache::getInstance()->getSpriteFrameByName(spriteFrameName);
-        CCASSERT(frame != NULL, "CCSpriteFrame must be non-NULL");
-
-        if (NULL == frame) return false;
-        bool pReturn = this->initWithSpriteFrame(frame, capInsets);
-        return pReturn;
+        bool ret = false;
+        do {
```

Comment:
why use do .. while(false) here? 
I see a lot of `do .. while(true)` use case, but `do ... while(false)` is seldom used. 
What's the advantage of `do ... while(false)`?